### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by MarcelMue**
Lot's of changes but all are just dependency updates. I tested this on `ginger` briefly and saw no issues.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by MarcelMue** [comment]
This is a preparation for #1261

File: go.mod [hunk 1]
```diff
@@ -4,18 +4,18 @@ go 1.14
 
 require (
 	github.com/ghodss/yaml v1.0.0
-	github.com/giantswarm/apiextensions/v2 v2.6.2
+	github.com/giantswarm/apiextensions/v3 v3.8.1-0.20201120151140-f03bde4123c1
```
Comment (calvix) [inline go.mod:7]: why using non released version here? aybe mistake?

Comment (MarcelMue) [reply go.mod:7]: Sorry forgot to mention - this is latest `apiextensions` master. I will tag it later on but wanted to create some PRs for cluster operator already.

**Review by njuettner — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Updated `apiextensions` and `operatorkit` dependencies.
```
Comment (stone-z) [inline CHANGELOG.md:14]:
```suggestion
- Update `apiextensions`, `k8sclient`, and `operatorkit` dependencies.
```

File: service/controller/resource/clusterconfigmap/resource.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ type Resource struct {
 // New creates a new configured config map state getter resource managing
 // cluster config maps.
 //
-//     https://godoc.org/github.com/giantswarm/operatorkit/v2/pkg/resource/k8s/secretresource#StateGetter
+//     https://godoc.org/github.com/giantswarm/operatorkit/v4/pkg/resource/k8s/secretresource#StateGetter
```
Comment (stone-z) [inline service/controller/resource/clusterconfigmap/resource.go:45]: This link does not work

File: service/controller/resource/encryptionkey/resource.go [hunk 4]
```diff
@@ -26,7 +26,7 @@ type Resource struct {
 // New creates a new configured secret state getter resource managing encryption
 // keys.
 //
-//     https://godoc.org/github.com/giantswarm/operatorkit/v2/pkg/resource/k8s/secretresource#StateGetter
+//     https://godoc.org/github.com/giantswarm/operatorkit/v4/pkg/resource/k8s/secretresource#StateGetter
```
Comment (stone-z) [inline service/controller/resource/encryptionkey/resource.go:29]: Same here

File: service/controller/resource/kubeconfig/resource.go [hunk 5]
```diff
@@ -36,7 +36,7 @@ type Resource struct {
 // New creates a new configured secret state getter resource managing kube
 // configs.
 //
-//     https://godoc.org/github.com/giantswarm/operatorkit/v2/pkg/resource/k8s/secretresource#StateGetter
+//     https://godoc.org/github.com/giantswarm/operatorkit/v4/pkg/resource/k8s/secretresource#StateGetter
```
Comment (stone-z) [inline service/controller/resource/kubeconfig/resource.go:39]: Same here

**Comment by stone-z** [comment]
Looks like the godocs have moved to https://pkg.go.dev/github.com/giantswarm/operatorkit

File: service/controller/resource/clusterconfigmap/resource.go (hunk 3 above)
Comment (MarcelMue) [reply service/controller/resource/clusterconfigmap/resource.go:45]: Neither does the v2 one :shrug:

Not sure what to do here.

**Comment by MarcelMue** [comment]
> Looks like the godocs have moved to https://pkg.go.dev/github.com/giantswarm/operatorkit

Should be fixed now :+1:

**Review by stone-z — APPROVED** [review]
I'm happy once `apiextensions` is tagged

**Review by rossf7 — APPROVED** [review]
LGTM once apiextensions is released.

**Comment by MarcelMue** [comment]
> LGTM once apiextensions is released.

BAM DONE

**Comment by MarcelMue** [comment]
Going with this

**Pull request merged**